### PR TITLE
Refactor `Accordion` to use a `getComponentConfiguration` helper to compute its configuration

### PR DESCRIPTION
**Pull request by patrickpatrickpatrick**
## What

- Move config specific helpers to new file (`configuration.mjs`)
- Change `extractConfigNamespace` parameters
- Add `GOVUKFrontendComponentConfigurable`
- Make `Accordion` extend `GOVUKFrontendComponentConfigurable`

## Why

As part of work to make configuration easier to use for custom components. 

Fixes https://github.com/alphagov/govuk-frontend/issues/5456

### CODE REVIEW

File: packages/govuk-frontend/src/govuk/common/configuration.mjs [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+ * Base Component class
+ *
+ * Centralises the behaviours shared by our components
+ *
+ * @virtual
+ * @template {ObjectNested} [ConfigurationType={}]
+ * @template {Element} [RootElementType=HTMLElement]
+ * @augments GOVUKFrontendComponent<RootElementType>
+ */
+export class GOVUKFrontendComponentConfigurable extends GOVUKFrontendComponent {
```
Comment (romaricpascal) [inline packages/govuk-frontend/src/govuk/common/configuration.mjs:16]: **nitpick** Something we can sort outside of the refactoring PRs is the naming of the class and whether we want to frontload with `Component` as it is now, or make it read like English with either `GOVUKFrontendConfigurableComponent` or `ConfigurableGOVUKFrontendComponent` 😊

File: packages/govuk-frontend/src/govuk/common/configuration.mjs [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+import { isObject, formatErrorMessage } from './index.mjs'
+
+/**
+ * Base Component class
+ *
+ * Centralises the behaviours shared by our components
+ *
+ * @virtual
+ * @template {ObjectNested} [ConfigurationType={}]
+ * @template {Element} [RootElementType=HTMLElement]
```
Comment (romaricpascal) [inline packages/govuk-frontend/src/govuk/common/configuration.mjs:13]: **question** If we were to update that type to `{Element & { dataset: DOMStringMap }}` would it remove the need to cast [on line 55](https://github.com/alphagov/govuk-frontend/pull/5475/files#diff-b2396d30ecca431162c287335bd70ad2314da293a2e8cfe33271a66828efee07R55)?

File: packages/govuk-frontend/src/govuk/common/configuration/extract-config-by-namespace.jsdom.test.mjs [hunk 3]
```diff
@@ -42,13 +42,17 @@ describe('extractConfigByNamespace', () => {
   it('defaults to empty config for known namespaces only', () => {
     const { dataset } = $element
 
-    const nonObject1 = extractConfigByNamespace(Component, dataset, 'a')
-    const nonObject2 = extractConfigByNamespace(Component, dataset, 'd')
-    const nonObject3 = extractConfigByNamespace(Component, dataset, 'e')
+    const nonObject1 = extractConfigByNamespace(Component.schema, dataset, 'a')
```
Comment (romaricpascal) [inline packages/govuk-frontend/src/govuk/common/configuration/extract-config-by-namespace.jsdom.test.mjs:45]: **suggestion** Seems we're only using the `Component` to access the `.schema` in this test file. That means we can drop the class in favour of just storing the `schema` in a variable to simplify the tests a little bit (I think we'll still need to annotate that variable with `@satisfies`).

**Review by romaricpascal — APPROVED** [review]
@patrickpatrickpatrick Looking good, thanks for carrying on after the pairing session. I've approved but noted 3 little things, 2 we could tweak before merging and one (the naming) I'll drop a message on Slack for discussion.

File: packages/govuk-frontend/src/govuk/common/configuration.mjs (hunk 2 above)
Comment (patrickpatrickpatrick) [reply packages/govuk-frontend/src/govuk/common/configuration.mjs]: Yes, we can! Nice spot.

**Pull request merged**